### ISSUE TITLE
click version conflict in metaflow and dependency installation failed

### ISSUE THREAD

**Issue by tianyi-qi-zefr** [Netflix/metaflow#1030]
hey guys, I need some help on version conflict in metaflow, one dependency `click` of `transformers` caused version conflict this morning, I tried to pin `sacremoses` version, it worked fine in local container but caused weird errors in metaflow. Appreciate any help! 

my dependency list 
```
# pip install 
metaflow==2.5.4
aioredis==1.3.1
tokenizers==0.10.1
sacremoses==0.0.41 # this does not fix error b 
transformers==4.4.2
ray[tune]==1.2.0
test-tube==0.7.5
aiohttp==3.5.4

# conda install 
pytorch-lightning==1.5.8
pytorch::pytorch==1.9.0
```

error when pinning sacremoses==0.0.41, this did not work 
```
<flow ContentRelevancyNN step train_model> failed:
    Internal error
Traceback (most recent call last):
  File "/metaflow/metaflow_ContentRelevancyNN_linux-64_458db741b217dd930030f43438417ad221462685/lib/python3.8/site-packages/pkg_resources/__init__.py", line 3031, in _dep_map
    return self.__dep_map
  File "/metaflow/metaflow_ContentRelevancyNN_linux-64_458db741b217dd930030f43438417ad221462685/lib/python3.8/site-packages/pkg_resources/__init__.py", line 2828, in __getattr__
    raise AttributeError(attr)
AttributeError: _DistInfoDistribution__dep_map
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/metaflow/metaflow_ContentRelevancyNN_linux-64_458db741b217dd930030f43438417ad221462685/lib/python3.8/site-packages/pkg_resources/__init__.py", line 3022, in _parsed_pkg_info
    return self._pkg_info
  File "/metaflow/metaflow_ContentRelevancyNN_linux-64_458db741b217dd930030f43438417ad221462685/lib/python3.8/site-packages/pkg_resources/__init__.py", line 2828, in __getattr__
    raise AttributeError(attr)
AttributeError: _pkg_info
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/metaflow/metaflow/cli.py", line 1102, in main
    start(auto_envvar_prefix="METAFLOW", obj=state)
  File "/metaflow/metaflow/_vendor/click/core.py", line 829, in __call__
    return self.main(args, kwargs)
  File "/metaflow/metaflow/_vendor/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/metaflow/metaflow/_vendor/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/metaflow/metaflow/_vendor/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, ctx.params)
  File "/metaflow/metaflow/_vendor/click/core.py", line 610, in invoke
    return callback(args, kwargs)
  File "/metaflow/metaflow/_vendor/click/decorators.py", line 21, in new_func
    return f(get_current_context(), args, kwargs)
  File "/metaflow/metaflow/cli.py", line 556, in step
    task.run_step(
  File "/metaflow/metaflow/task.py", line 547, in run_step
    self._exec_step_function(step_func)
  File "/metaflow/metaflow/task.py", line 53, in _exec_step_function
    step_function()
  File "/metaflow/decorators/pip_wrapper.py", line 37, in wrapper
    return function(args, kwargs)
  File "train_cr_yt.py", line 766, in train_model
    from cr_steps.train_model_step import train_model_step, TrainModelStepOutput
  File "/metaflow/cr_steps/train_model_step.py", line 9, in <module>
    import pytorch_lightning as pl
  File "/metaflow/metaflow_ContentRelevancyNN_linux-64_458db741b217dd930030f43438417ad221462685/lib/python3.8/site-packages/pytorch_lightning/__init__.py", line 20, in <module>
    from pytorch_lightning.callbacks import Callback  # noqa: E402
  File "/metaflow/metaflow_ContentRelevancyNN_linux-64_458db741b217dd930030f43438417ad221462685/lib/python3.8/site-packages/pytorch_lightning/callbacks/__init__.py", line 14, in <module>
    from pytorch_lightning.callbacks.base import Callback
  File "/metaflow/metaflow_ContentRelevancyNN_linux-64_458db741b217dd930030f43438417ad221462685/lib/python3.8/site-packages/pytorch_lightning/callbacks/base.py", line 26, in <module>
    from pytorch_lightning.utilities.types import STEP_OUTPUT
  File "/metaflow/metaflow_ContentRelevancyNN_linux-64_458db741b217dd930030f43438417ad221462685/lib/python3.8/site-packages/pytorch_lightning/utilities/types.py", line 25, in <module>
    from torchmetrics import Metric
  File "/metaflow/metaflow_ContentRelevancyNN_linux-64_458db741b217dd930030f43438417ad221462685/lib/python3.8/site-packages/torchmetrics/__init__.py", line 14, in <module>
    from torchmetrics import functional  # noqa: E402
  File "/metaflow/metaflow_ContentRelevancyNN_linux-64_458db741b217dd930030f43438417ad221462685/lib/python3.8/site-packages/torchmetrics/functional/__init__.py", line 14, in <module>
    from torchmetrics.functional.audio.pit import permutation_invariant_training, pit_permutate
  File "/metaflow/metaflow_ContentRelevancyNN_linux-64_458db741b217dd930030f43438417ad221462685/lib/python3.8/site-packages/torchmetrics/functional/audio/__init__.py", line 14, in <module>
    from torchmetrics.functional.audio.pit import permutation_invariant_training, pit_permutate  # noqa: F401
  File "/metaflow/metaflow_ContentRelevancyNN_linux-64_458db741b217dd930030f43438417ad221462685/lib/python3.8/site-packages/torchmetrics/functional/audio/pit.py", line 21, in <module>
    from torchmetrics.utilities.imports import _SCIPY_AVAILABLE
  File "/metaflow/metaflow_ContentRelevancyNN_linux-64_458db741b217dd930030f43438417ad221462685/lib/python3.8/site-packages/torchmetrics/utilities/imports.py", line 115, in <module>
    _TRANSFORMERS_AUTO_AVAILABLE = _module_available("transformers.models.auto")
  File "/metaflow/metaflow_ContentRelevancyNN_linux-64_458db741b217dd930030f43438417ad221462685/lib/python3.8/site-packages/torchmetrics/utilities/imports.py", line 57, in _module_available
    module = import_module(module_names[0])
  File "/metaflow/metaflow_ContentRelevancyNN_linux-64_458db741b217dd930030f43438417ad221462685/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "/metaflow/metaflow_ContentRelevancyNN_linux-64_458db741b217dd930030f43438417ad221462685/lib/python3.8/site-packages/transformers/__init__.py", line 43, in <module>
    from . import dependency_versions_check
  File "/metaflow/metaflow_ContentRelevancyNN_linux-64_458db741b217dd930030f43438417ad221462685/lib/python3.8/site-packages/transformers/dependency_versions_check.py", line 41, in <module>
    require_version_core(deps[pkg])
  File "/metaflow/metaflow_ContentRelevancyNN_linux-64_458db741b217dd930030f43438417ad221462685/lib/python3.8/site-packages/transformers/utils/versions.py", line 94, in require_version_core
    return require_version(requirement, hint)
  File "/metaflow/metaflow_ContentRelevancyNN_linux-64_458db741b217dd930030f43438417ad221462685/lib/python3.8/site-packages/transformers/utils/versions.py", line 80, in require_version
    got_ver = pkg_resources.get_distribution(pkg).version
  File "/metaflow/metaflow_ContentRelevancyNN_linux-64_458db741b217dd930030f43438417ad221462685/lib/python3.8/site-packages/pkg_resources/__init__.py", line 471, in get_distribution
    dist = get_provider(dist)
  File "/metaflow/metaflow_ContentRelevancyNN_linux-64_458db741b217dd930030f43438417ad221462685/lib/python3.8/site-packages/pkg_resources/__init__.py", line 347, in get_provider
    return working_set.find(moduleOrReq) or require(str(moduleOrReq))[0]
  File "/metaflow/metaflow_ContentRelevancyNN_linux-64_458db741b217dd930030f43438417ad221462685/lib/python3.8/site-packages/pkg_resources/__init__.py", line 891, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/metaflow/metaflow_ContentRelevancyNN_linux-64_458db741b217dd930030f43438417ad221462685/lib/python3.8/site-packages/pkg_resources/__init__.py", line 785, in resolve
    new_requirements = dist.requires(req.extras)[::-1]
  File "/metaflow/metaflow_ContentRelevancyNN_linux-64_458db741b217dd930030f43438417ad221462685/lib/python3.8/site-packages/pkg_resources/__init__.py", line 2749, in requires
    dm = self._dep_map
  File "/metaflow/metaflow_ContentRelevancyNN_linux-64_458db741b217dd930030f43438417ad221462685/lib/python3.8/site-packages/pkg_resources/__init__.py", line 3033, in _dep_map
    self.__dep_map = self._compute_dependencies()
  File "/metaflow/metaflow_ContentRelevancyNN_linux-64_458db741b217dd930030f43438417ad221462685/lib/python3.8/site-packages/pkg_resources/__init__.py", line 3042, in _compute_dependencies
    for req in self._parsed_pkg_info.get_all('Requires-Dist') or []:
  File "/metaflow/metaflow_ContentRelevancyNN_linux-64_458db741b217dd930030f43438417ad221462685/lib/python3.8/site-packages/pkg_resources/__init__.py", line 3024, in _parsed_pkg_info
    metadata = self.get_metadata(self.PKG_INFO)
  File "/metaflow/metaflow_ContentRelevancyNN_linux-64_458db741b217dd930030f43438417ad221462685/lib/python3.8/site-packages/pkg_resources/__init__.py", line 1412, in get_metadata
    value = self._get(path)
  File "/metaflow/metaflow_ContentRelevancyNN_linux-64_458db741b217dd930030f43438417ad221462685/lib/python3.8/site-packages/pkg_resources/__init__.py", line 1616, in _get
    with open(path, 'rb') as stream:
FileNotFoundError: [Errno 2] No such file or directory: '/metaflow/metaflow_ContentRelevancyNN_linux-64_458db741b217dd930030f43438417ad221462685/lib/python3.8/site-packages/click-7.1.2.dist-info/METADATA'
```

error when removing sacremoses==0.0.41 in pip install, this stopped working since this morning as `click` version is pinned on https://github.com/alvations/sacremoses/blob/master/setup.py#L23 today. 
```
<flow ContentRelevancyNN step train_model> failed:
    Internal error
Traceback (most recent call last):
  File "/metaflow/metaflow/cli.py", line 1102, in main
    start(auto_envvar_prefix="METAFLOW", obj=state)
  File "/metaflow/metaflow/_vendor/click/core.py", line 829, in __call__
    return self.main(args, kwargs)
  File "/metaflow/metaflow/_vendor/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/metaflow/metaflow/_vendor/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/metaflow/metaflow/_vendor/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, ctx.params)
  File "/metaflow/metaflow/_vendor/click/core.py", line 610, in invoke
    return callback(args, kwargs)
  File "/metaflow/metaflow/_vendor/click/decorators.py", line 21, in new_func
    return f(get_current_context(), args, kwargs)
  File "/metaflow/metaflow/cli.py", line 556, in step
    task.run_step(
  File "/metaflow/metaflow/task.py", line 547, in run_step
    self._exec_step_function(step_func)
  File "/metaflow/metaflow/task.py", line 53, in _exec_step_function
    step_function()
  File "/metaflow/decorators/pip_wrapper.py", line 37, in wrapper
    return function(args, kwargs)
  File "train_cr_yt.py", line 766, in train_model
    from cr_steps.train_model_step import train_model_step, TrainModelStepOutput
  File "/metaflow/cr_steps/train_model_step.py", line 9, in <module>
    import pytorch_lightning as pl
  File "/metaflow/metaflow_ContentRelevancyNN_linux-64_458db741b217dd930030f43438417ad221462685/lib/python3.8/site-packages/pytorch_lightning/__init__.py", line 20, in <module>
    from pytorch_lightning.callbacks import Callback  # noqa: E402
  File "/metaflow/metaflow_ContentRelevancyNN_linux-64_458db741b217dd930030f43438417ad221462685/lib/python3.8/site-packages/pytorch_lightning/callbacks/__init__.py", line 14, in <module>
    from pytorch_lightning.callbacks.base import Callback
  File "/metaflow/metaflow_ContentRelevancyNN_linux-64_458db741b217dd930030f43438417ad221462685/lib/python3.8/site-packages/pytorch_lightning/callbacks/base.py", line 26, in <module>
    from pytorch_lightning.utilities.types import STEP_OUTPUT
  File "/metaflow/metaflow_ContentRelevancyNN_linux-64_458db741b217dd930030f43438417ad221462685/lib/python3.8/site-packages/pytorch_lightning/utilities/types.py", line 25, in <module>
    from torchmetrics import Metric
  File "/metaflow/metaflow_ContentRelevancyNN_linux-64_458db741b217dd930030f43438417ad221462685/lib/python3.8/site-packages/torchmetrics/__init__.py", line 14, in <module>
    from torchmetrics import functional  # noqa: E402
  File "/metaflow/metaflow_ContentRelevancyNN_linux-64_458db741b217dd930030f43438417ad221462685/lib/python3.8/site-packages/torchmetrics/functional/__init__.py", line 14, in <module>
    from torchmetrics.functional.audio.pit import permutation_invariant_training, pit_permutate
  File "/metaflow/metaflow_ContentRelevancyNN_linux-64_458db741b217dd930030f43438417ad221462685/lib/python3.8/site-packages/torchmetrics/functional/audio/__init__.py", line 14, in <module>
    from torchmetrics.functional.audio.pit import permutation_invariant_training, pit_permutate  # noqa: F401
  File "/metaflow/metaflow_ContentRelevancyNN_linux-64_458db741b217dd930030f43438417ad221462685/lib/python3.8/site-packages/torchmetrics/functional/audio/pit.py", line 21, in <module>
    from torchmetrics.utilities.imports import _SCIPY_AVAILABLE
  File "/metaflow/metaflow_ContentRelevancyNN_linux-64_458db741b217dd930030f43438417ad221462685/lib/python3.8/site-packages/torchmetrics/utilities/imports.py", line 115, in <module>
    _TRANSFORMERS_AUTO_AVAILABLE = _module_available("transformers.models.auto")
  File "/metaflow/metaflow_ContentRelevancyNN_linux-64_458db741b217dd930030f43438417ad221462685/lib/python3.8/site-packages/torchmetrics/utilities/imports.py", line 57, in _module_available
    module = import_module(module_names[0])
  File "/metaflow/metaflow_ContentRelevancyNN_linux-64_458db741b217dd930030f43438417ad221462685/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "/metaflow/metaflow_ContentRelevancyNN_linux-64_458db741b217dd930030f43438417ad221462685/lib/python3.8/site-packages/transformers/__init__.py", line 43, in <module>
    from . import dependency_versions_check
  File "/metaflow/metaflow_ContentRelevancyNN_linux-64_458db741b217dd930030f43438417ad221462685/lib/python3.8/site-packages/transformers/dependency_versions_check.py", line 41, in <module>
    require_version_core(deps[pkg])
  File "/metaflow/metaflow_ContentRelevancyNN_linux-64_458db741b217dd930030f43438417ad221462685/lib/python3.8/site-packages/transformers/utils/versions.py", line 94, in require_version_core
    return require_version(requirement, hint)
  File "/metaflow/metaflow_ContentRelevancyNN_linux-64_458db741b217dd930030f43438417ad221462685/lib/python3.8/site-packages/transformers/utils/versions.py", line 80, in require_version
    got_ver = pkg_resources.get_distribution(pkg).version
  File "/metaflow/metaflow_ContentRelevancyNN_linux-64_458db741b217dd930030f43438417ad221462685/lib/python3.8/site-packages/pkg_resources/__init__.py", line 471, in get_distribution
    dist = get_provider(dist)
  File "/metaflow/metaflow_ContentRelevancyNN_linux-64_458db741b217dd930030f43438417ad221462685/lib/python3.8/site-packages/pkg_resources/__init__.py", line 347, in get_provider
    return working_set.find(moduleOrReq) or require(str(moduleOrReq))[0]
  File "/metaflow/metaflow_ContentRelevancyNN_linux-64_458db741b217dd930030f43438417ad221462685/lib/python3.8/site-packages/pkg_resources/__init__.py", line 891, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/metaflow/metaflow_ContentRelevancyNN_linux-64_458db741b217dd930030f43438417ad221462685/lib/python3.8/site-packages/pkg_resources/__init__.py", line 782, in resolve
    raise VersionConflict(dist, req).with_context(dependent_req)
pkg_resources.ContextualVersionConflict: (click 7.1.2 (/metaflow/metaflow_ContentRelevancyNN_linux-64_458db741b217dd930030f43438417ad221462685/lib/python3.8/site-packages), Requirement.parse('click==8.0'), {'sacremoses'})
```

**Comment by savingoyal** [comment]
@tianyi-qi-zefr What version of Metaflow are you using?

**Comment by tianyi-qi-zefr** [comment]
> @tianyi-qi-zefr What version of Metaflow are you using?

updated in dependency, it is 2.5.4

**Comment by savingoyal** [comment]
Follow-up thread - https://outerbounds-community.slack.com/archives/C02116BBNTU/p1651533302035489 accessible through slack.outerbounds.co

It seems that manually pip installing packages within a conda environment generated through Metaflow is breaking the environment.

**Comment by tianyi-qi-zefr** [comment]
issue fixed when pinning click==7.1.2
